js/main.js

```diff
@@ -126,9 +155,16 @@ ${showButtons(post, canUserLikePost)}
 
 function show() {
 	const postToShow = allPosts
+  .filter(post => {
+    if (selectValue === "#All")
+      return true
+    if (idToUsername(post.ownerId) === selectValue) {
+      return true
+    }
```

Comment:
How about?
```suggestion
    const shouldShowAllPosts = selectValue === '#All';
    const shouldShowMyPosts = idToUsername(post.ownerId) === selectValue;

    return shouldShowAllPosts || shouldShowMyPosts;
```

But the wonderful solution will be include all the logic inside one `filter`